app/webpacker/controllers/trixeditor_controller.js

```diff
@@ -2,9 +2,9 @@ import { Controller } from "stimulus";
 
 export default class extends Controller {
   connect() {
-    window.addEventListener("trix-change", this.#trixChange);
+    this.element.addEventListener("trix-change", this.#trixChange);
     this.#trixInitialize();
-    window.addEventListener("trix-initialize", this.#trixInitialize);
+    this.element.addEventListener("trix-initialize", this.#trixInitialize);
```

Comment:
We also need to remove the added event in `disconnect` method.

Reply:
Not sure I fully follow here! (Can't find the disconnect function in this controller)

Do you mean reverting to attach the event to the global window object?

Reply:
I'm not 100% sure, but when this stimulus component is removed (eg if the section was reloaded with a Turbo Stream or similar), the element (`this.element`) is removed from the DOM, and the browser would clean up any attached event listeners. 

I only use `disconnect` when the controller has changed something outside this DOM element (eg the global event listener). 

Anyway, it won't hurt to explicitly clean it up just in case 👍

Reply:
In this scenario, the removal of the added event listeners to `this.element` is handled by the browser by default when the element is removed from the page.
Also, I agree :100:  that we explicitly remove added events to prevent any uncertainty about the event not properly removed